src/plugin/electron/settingsimportui.js

```diff
@@ -0,0 +1,212 @@
+ * @param {!ElectronOS.SettingsFile} existing The existing file config.
+ */
+const confirmReplaceLocal = (fileName, content, existing) => {
+  const prompt = `The imported settings file already exists with the label "${existing.label}". Would you like to ` +
+      'replace the existing file, or save as a new file?';
+  launchConfirm({
+    confirm: () => {
+      // Replace the existing settings file.
+      confirmLabel(fileName, content, existing.label);
+    },
+    cancel: () => {
+      const newName = getUniqueSettingsFile(fileName);
+      confirmLabel(newName, content);
+    },
+    prompt,
+    yesText: 'Replace',
+    yesIcon: 'fas fa-exchange-alt',
+    yesButtonClass: 'btn-danger',
+    noText: 'Save as New',
+    noIcon: 'far fa-save',
+    noButtonClass: 'btn-secondary',
+    windowOptions: /** @type {!osx.window.WindowOptions} */ ({
+      icon: 'fas fa-cogs',
+      label: 'Settings File Already Exists',
+      showClose: true
+    })
+  });
+};
+
+
+/**
+ * Confirm if an existing remote settings file should be replaced.
+ * @param {string} url The file URL.
+ * @param {string} fileName The file name.
+ * @param {!ElectronOS.SettingsFile} existing The existing file config.
+ */
+const confirmReplaceUrl = (url, fileName, existing) => {
+  const prompt = `The imported settings file already exists with the label "${existing.label}". Would you like to ` +
+      'replace the existing file?';
+  launchConfirm({
+    confirm: () => {
+      // Replace the existing settings file.
+      confirmLabel(url, null, existing.label);
+    },
+    prompt,
+    yesText: 'Replace',
+    yesIcon: 'fas fa-exchange-alt',
+    yesButtonClass: 'btn-danger',
+    windowOptions: /** @type {!osx.window.WindowOptions} */ ({
+      icon: 'fas fa-cogs',
+      label: 'Settings File Already Exists',
+      showClose: true
+    })
+  });
+};
+
+
+/**
+ * Get a unique settings file name.
+ * @param {string} fileName The original file name.
+ * @return {string} The unique name.
+ */
+const getUniqueSettingsFile = (fileName) => {
+  const baseName = fileName.replace(/\.[^.]+$/, '');
+
+  let i = 1;
+  let newName = `${baseName}-${i}.json`;
+  while (ElectronOS.getSettingsFile(newName)) {
+    newName = `${baseName}-${++i}.json`;
+  }
+
+  return newName;
+};
+
+
+/**
+ * UI to import settings files.
+ */
+export default class SettingsImportUI extends FileImportUI {
+  /**
+   * Constructor.
+   */
+  constructor() {
+    super();
+  }
+
+  /**
+   * @inheritDoc
+   */
+  launchUI(file, opt_config) {
+    file.convertContentToString();
+
+    const content = /** @type {string} */ (file.getContent());
+    const url = file.getUrl();
+
+    // Internal settings files are prefixed with a '.', so remove those from the filename if present to avoid conflicts.
+    const fileName = file.getFileName().replace(/^\.+/, '');
+
+    if (content && url) {
+      try {
+        const parsed = JSON.parse(content);
+        if (parsed && jsonMime.detect(null, file, parsed)) {
```

Comment:
Is there a reason we need to check the file type here again? Shouldn't that be done by the `ImportManager` stack before ever launching this UI?

Reply:
I want to say there was a path where the content wasn't checked yet so I was verifying it appeared to be a JSON settings file. I'll have to take a look.

Reply:
Ah right, this was for drag/drop which doesn't go through the `ImportManager`. The drop is handled by the UI directly via the `url-drag-drop` directive.

Reply:
Gotcha, yeah the drag drop directive design is a bit old and jank, might be something to look at in the future.